### PR TITLE
Compaction reason

### PR DESCRIPTION
**Pull request by zhangjinpeng1987**
## What have you changed? (mandatory)

Add a metric to monitor the compaction reason, so we can know how many manual compactions have been triggered.

## What are the type of the changes? (mandatory)

The currently defined types are listed below, please pick one of the types for this PR by removing the others:

Improvement

## How has this PR been tested? (mandatory)

No

## Does this PR affect documentation (docs/docs-cn) update? (mandatory)

No

## Does this PR affect tidb-ansible update? (mandatory)

This pr will not affect existing metrics, i will send a pr to display this metric for tidb-ansible.

### CODE REVIEW

**Comment by siddontang** [comment]
Do you add it to the Grafana?

File: src/util/rocksdb/event_listener.rs [hunk 1]
```diff
@@ -65,6 +65,13 @@ impl rocksdb::EventListener for EventListener {
         STORE_ENGINE_COMPACTION_NUM_CORRUPT_KEYS_VEC
             .with_label_values(&[&self.db_name, info.cf_name()])
             .inc_by(info.num_corrupt_keys() as i64);
+        STORE_ENGINE_COMPACTION_REASON_VEC
+            .with_label_values(&[
+                &self.db_name,
+                info.cf_name(),
+                &info.compaction_reason().to_string(),
```
Comment (siddontang) [inline src/util/rocksdb/event_listener.rs]: em, need `to_string` here?

Comment (zhangjinpeng1987) [reply src/util/rocksdb/event_listener.rs]: CompactionReason is a enum type which implement fmt::Display, so need to_string. And a compaction job only call this function once, so it is not frequency operation.

**Comment by zhangjinpeng1987** [comment]
> Do you add it to the Grafana?

As mentioned in description, i will add it to the Grafana after this pr has merged.

**Pull request merged**